realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -897,24 +897,139 @@ JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeNotEqual__J_3JLja
     TableQuery_StringPredicate(env, nativeQueryPtr, columnIndexes, value, caseSensitive, StringNotEqual);
 }
 
-JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeBeginsWith(
+JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeBeginsWith__J_3JLjava_lang_String_2Z(
```

Comment:
hmmmm, interesting ...  from jni.h
```
class _jstring : public _jobject {};
class _jarray : public _jobject {};
class _jbyteArray : public _jarray {};
```

The `jstring` and `jbytearray` both have the same type `jobject`. So the method names have to be different.